### PR TITLE
Isolated Server Persistence

### PR DESCRIPTION
**Pull request by KaustubhShamshery**
## Backward Compatibility

- [x] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [x] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

**Review by bb111189 — APPROVED** [review]

File: src/cmd/isolated_server.cpp [hunk 1]
```diff
@@ -128,25 +135,34 @@ int main(int argc, const char* argv[]) {
...
+    AccountStore::GetInstance().InitSoft();
 
-    mediator.RegisterColleagues(nullptr, &node, &lk, vd.get());
+    uint64_t blocknum;
 
-    AccountStore::GetInstance().Init();
-    if (readAccountJsonFromFile(accountJsonFilePath)) {
-      cerr << "ERROR: "
-           << "Unable to parse account json file" << endl;
+    if (!loadPersistence && accountJsonFilePath.empty()) {
```
Comment (ansnunez) [inline src/cmd/isolated_server.cpp:144]: What if I want to launch isolated server from clean slate?

File: src/cmd/isolated_server.cpp [hunk 2]
```diff
@@ -159,6 +175,14 @@ int main(int argc, const char* argv[]) {
       cout << "Server listening on " << port << endl;
     }
 
+    if (loadPersistence) {
```
Comment (ansnunez) [inline src/cmd/isolated_server.cpp:178]: Probably better to do this before starting to listen on port?

File: src/libServer/IsolatedServer.cpp [hunk 3]
```diff
@@ -191,6 +193,30 @@ bool IsolatedServer::ValidateTxn(const Transaction& tx, const Address& fromAddr,
...
 
+bool IsolatedServer::RetrieveHistory() {
+  m_mediator.m_txBlockChain.Reset();
+
+  std::shared_ptr<Retriever> m_retriever;
+
+  bool st_result = m_retriever->RetrieveStates();
+
+  if (!(st_result)) {
+    LOG_GENERAL(WARNING, "Retrieval of states and tx block failed ");
```
Comment (ansnunez) [inline src/libServer/IsolatedServer.cpp:204]:
```suggestion
    LOG_GENERAL(WARNING, "Retrieval of states and tx block failed");
```

File: src/libServer/IsolatedServer.cpp [hunk 4]
```diff
@@ -342,26 +370,44 @@ Json::Value IsolatedServer::CreateTransaction(const Json::Value& _json) {
...
-  lock_guard<mutex> g(m_txnBlockNumMapMutex);
-
-  const auto& iter = m_txnBlockNumMap.find(txNum);
+  auto const& txBlock = m_mediator.m_txBlockChain.GetBlock(txNum);
 
-  if (iter == m_txnBlockNumMap.end()) {
-    throw JsonRpcException(RPC_INVALID_PARAMETER, "No txns found");
+  if (txBlock.GetHeader().GetBlockNum() == INIT_BLOCK_NUMBER &&
+      txBlock.GetHeader().GetDSBlockNum() == INIT_BLOCK_NUMBER) {
+    throw JsonRpcException(RPC_INVALID_PARAMS, "Tx Block does not exist");
```
Comment (ansnunez) [inline src/libServer/IsolatedServer.cpp:383]:
```suggestion
    throw JsonRpcException(RPC_INVALID_PARAMS, "TxBlock does not exist");
```

Comment (ansnunez) [reply src/libServer/IsolatedServer.cpp:383]: For consistency with line 409

**Review by ansnunez — CHANGES_REQUESTED** [review]

File: src/cmd/isolated_server.cpp (hunk 1 above)
Comment (KaustubhShamshery) [reply src/cmd/isolated_server.cpp:144]: then just use the `-f` version

**Review by bb111189 — APPROVED** [review]

**Review by ansnunez — APPROVED** [review]

**Pull request merged**